tini/src/seccomp_fd_notify.c

```diff
@@ -217,213 +226,130 @@ struct sock_filter net_perf_filter[] = {
-			PRINT_INFO("Intercepted sendmsg - passthrough!\n");
-			set_break = 1;
-		} else {
-			PRINT_INFO("Intercepted syscall %d - passthrough!\n", req->data.nr);
-		}
-		set_seccomp_response_to_continue(req, resp);
-		respond_to_seccomp_client(notify_fd, resp);
-		if (set_break) {
+	socket_path = getenv(TITUS_SECCOMP_NOTIFY_SOCK_PATH);
+	if (!socket_path) {
+		PRINT_WARNING("TITUS_SECCOMP_NOTIFY_SOCK_PATH not defined.");
+		kill(child_pid, SIGKILL);
+		return;
+	}
+	int sock_fd = -1;
+	// Sometimes things are not perfect, and the socket is not ready at first
+	// Instead of enforcing strict ordering, we can be defensive and retry.
+	int attempts = 10;
+	for (int i = 1; i <= attempts; i++) {
+		sock_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+		if (sock_fd != -1) {
 			break;
 		}
+		PRINT_INFO(
+			"Titus seccomp unix socket not ready on attempt %d/%d, Sleeping 1 second",
+			i, attempts);
+		sleep(1);
+	}
+	if (sock_fd == -1) {
+		PRINT_WARNING(
+			"Unable to open unix socket for seccomp handoff after %d attempts: %s",
+			attempts, strerror(errno));
+		kill(child_pid, SIGKILL);
+		return;
 	}
-	PRINT_INFO("Stopped polling notifyfd on Tini, over to TSA!");
-	return NULL;
-}
-
-void maybe_setup_seccomp_notifer() {
-	char *socket_path;
-	pthread_t thread_id;
-
-	socket_path = getenv(TITUS_SECCOMP_NOTIFY_SOCK_PATH);
-	if (socket_path) {
-		int sock_fd = -1;
-		// Sometimes things are not perfect, and the socket is not ready at first
-		// Instead of enforcing strict ordering, we can be defensive and retry.
-		int attempts = 10;
-		for (int i = 1; i <= attempts; i++) {
-			sock_fd = socket(AF_UNIX, SOCK_STREAM, 0);
-			if (sock_fd != -1) {
-				break;
-			}
-			PRINT_INFO(
-				"Titus seccomp unix socket not ready on attempt %d/%d, Sleeping 1 second",
-				i, attempts);
-			sleep(1);
-		}
-		if (sock_fd == -1) {
-			PRINT_WARNING(
-				"Unable to open unix socket for seccomp handoff after %d attempts: %s",
-				attempts, strerror(errno));
-			return;
-		}
-
-		struct sockaddr_un addr = { 0 };
-		addr.sun_family = AF_UNIX;
-		strncpy(addr.sun_path, socket_path, sizeof(addr.sun_path) - 1);
-		int result = -1;
-		for (int i = 1; i <= attempts; i++) {
-			result = connect(sock_fd, (struct sockaddr *)&addr,
-					 sizeof(addr));
-			if (result != -1) {
-				break;
-			}
-			PRINT_INFO(
-				"Titus seccomp unix socket not connectable yet on attempt %d/%d, Sleeping 1 second",
-				i, attempts);
-			sleep(1);
-		}
-		if (result == -1) {
-			PRINT_WARNING(
-				"Unable to connect on unix socket (%s) for seccomp handoff after %d attempts: %s",
-				socket_path, attempts, strerror(errno));
-			return;
-		}
 
-		int notify_fd = -1;
-		notify_fd = install_notify_filter();
-		if (notify_fd == -1) {
-			close(sock_fd);
-			return;
+	struct sockaddr_un addr = { 0 };
+	addr.sun_family = AF_UNIX;
+	strncpy(addr.sun_path, socket_path, sizeof(addr.sun_path) - 1);
+	int result = -1;
+	for (int i = 1; i <= attempts; i++) {
+		result = connect(sock_fd, (struct sockaddr *)&addr,
```

Comment:
This might be a dumb question, but why isn't this done in the child?

Reply:
Ah, it could be  - just wrote the code out trying to isolate the child to do the single task of just sending over the fd (or call sendmsg) only and the parent setting up the `fds` in the  `struct tsa_fds`.